fluidkeys/secretsend.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/fluidkeys/fluidkeys/colour"
+
+	"github.com/fluidkeys/crypto/openpgp"
+	"github.com/fluidkeys/crypto/openpgp/armor"
+	"github.com/fluidkeys/fluidkeys/api"
+	"github.com/fluidkeys/fluidkeys/out"
+	"github.com/fluidkeys/fluidkeys/pgpkey"
+)
+
+func secretSend(recipientEmail string) exitCode {
+	out.Print("\n")
+	out.Print("[type or paste your message, ending by typing Ctrl-D]\n\n")
+
+	secret, err := scanUntilEOF()
+	if err != nil {
+		panic(err)
+		return 1
+	}
+
+	client := api.NewClient()
+	armoredPublicKey, _, err := client.GetPublicKey(recipientEmail)
+	if err != nil {
+		printFailed("Couldn't get the public key for " + recipientEmail + "\n")
+	}
+
+	printSuccess("Found public key for " + recipientEmail + "\n")
+
+	pgpKey, err := pgpkey.LoadFromArmoredPublicKey(armoredPublicKey)
+	if err != nil {
+		printFailed("Couldn't load the public key:")
+		out.Print("Error: " + err.Error() + "\n")
+		return 1
+	}
+
+	encryptedSecret, err := encryptSecret(secret, pgpKey)
+	if err != nil {
+		printFailed("Couldn't encrypt the secret:")
+		out.Print("Error: " + err.Error() + "\n")
+		return 1
+	}
+
+	response, err := client.CreateSecret(fmt.Sprintf("OPENPGP4FPR:%s", pgpKey.Fingerprint().Hex()), encryptedSecret)
+	if response.StatusCode != 201 || err != nil {
+		printFailed("Couldn't send the secret to " + recipientEmail)
+		if err != nil {
```

Comment:
Worth noting: if the API returns a 400, 500, whatever, I've tried hard to ensure it always includes a JSON-encoded human friendly message:
```json
{
    "detail": "the database didn't respond"
}
```

... you could make a helper which tries to decode that (see `v1structs.ErrorResponse`) and possible display it